qotd/blog.md

```diff
@@ -0,0 +1,249 @@
+
+```py
+# use the image we downloaded in the above step
+img = Image.open("img.png")
+width, height = img.size
+image_editable = ImageDraw.Draw(img)
+
+# wrap text
+lines = textwrap.wrap(text, width=40)
+
+# get the line count and generate a starting offset on y-axis
+line_count = len(lines)
+y_offset = height/2 - (line_count/2 * title_font.getbbox(lines[0])[3])
+
+# for each line of text, we generate a (x,y) to calculate the poisitioning
+for line in lines:
+    (_, _, line_w, line_h) = title_font.getbbox(line)
+    x = (width - line_w)/2
+    image_editable.text((x,y_offset), line, (237, 230, 211), font=title_font)
+    y_offset += line_h
+img.save("result.jpg")
+print("generated " + filename)
+return filename
+```
+
+This generates the final image called `result.jpg`
+
+#### Uploading the image
+For the penultimate step, we need to upload the image so that we can use that with Courier. In this case, I'm using Firebase Storage but you can feel free to use whatever you like.
+
+```py
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import storage
+
+cred = credentials.Certificate('serviceaccount.json')
+firebase_admin.initialize_app(cred, {...})
+
+bucket = storage.bucket()
+blob = bucket.blob(filename)
+blob.upload_from_filename(filename)
+blob.make_public()
+return blob.public_url
+```
+
+## Step 3: Integrating with Courier
+
+Finally, we have everything we need to start sending our awesome quotes to our friends and family. We can use Courier to create a good looking email template.
+
+#### Creating the template on courier
+
+![img](https://i.imgur.com/0Ktbw0E.png)
+
+#### Sending the message
+
+Sending a message with Courier is as easy it gets. While courier has its own SDKs that can make integration easy, I prefer using their API endpoint to keep things simple. With my `AUTH_TOKEN` and `TEMPLATE_ID` in hand, we can use the following piece of code to send our image
+
+```py
+import requests
+
+headers = {
+    "Accept": "application/json",
+    "Content-Type": "application/json",
+    "Authorization": "Bearer {}".format(os.environ['COURIER_AUTH_TOKEN'])
+}
+message={
+    "to": { "email": os.environ["COURIER_RECIPIENT"] },
+    "data": {
+        "date": datetime.today().strftime("%B %d, %Y"),
+        "img": image_url ## this is image url we generated earlier
+    },
+    "routing": {
+        "method": "single",
+        "channels": [
+            "email"
+        ]
+    },
+    "template": os.environ["COURIER_TEMPLATE"]
+}
+requests.post("https://api.courier.com/send", json={"message": message}, headers=headers)
+```
+
+And that's it!
+
+## Conclusions
+
+This tutorial demonstrated how easy it is to get started with machine learning & Courier.
+
+If you want to go ahead and improve this project, here are some interesting ideas to try
+
+- Better background image: Use a term from the generated quote to search for an image?
```

Comment:
This is awesome! Overall, great blogpost. I'm sure our community would love to build this project themselves.

Reply:
Thank you! Glad to hear that!